### PR TITLE
3.3 to main

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Merges:
- https://github.com/juju/juju/pull/16054
- https://github.com/juju/juju/pull/16062
- https://github.com/juju/juju/pull/16087
- https://github.com/juju/juju/pull/16088
- https://github.com/juju/juju/pull/16079

Conflicts:
- apiserver/facades/agent/secretsmanager/package_test.go
- apiserver/facades/agent/secretsmanager/register.go
- apiserver/facades/agent/secretsmanager/secrets.go
- apiserver/facades/agent/secretsmanager/secrets_test.go
- cmd/juju/application/deploy_test.go
- cmd/juju/application/deployer/deployer_test.go
- cmd/juju/application/export_test.go
- cmd/juju/application/refresh_test.go
- cmd/juju/application/unexpose_test.go
- cmd/juju/machine/upgrademachine_test.go
- core/base/supported.go
- core/base/supportedbases_test.go
- core/base/supportedseries.go
- core/base/supportedseries_linux_test.go
- core/base/supportedseries_test.go
- core/os/os.go
- worker/uniter/runner/context/env_test.go

All conflicts resulted from either https://github.com/juju/juju/pull/16062 (mostly this) or https://github.com/juju/juju/pull/16079. They were all _fairly_ trivial to solve, but I would want @ycliuhw to have a look over his conflicts if possible

### CODE REVIEW

File: apiserver/facades/agent/secretsmanager/secrets.go [hunk 1]
```diff
@@ -26,10 +27,7 @@ import (
 	"github.com/juju/juju/state"
 )
 
-// For testing.
-var (
-	GetProvider = secretsprovider.Provider
-)
+var logger = loggo.GetLoggerWithLabels("juju.apiserver.secretsmanager", corelogger.SECRETS)
```
Comment (ycliuhw) [inline apiserver/facades/agent/secretsmanager/secrets.go:30]: I don't think we need this. Coz we have SecretsManagerAPI.logger.

**Review by ycliuhw — APPROVED** [review]
LGTM, thank you!

File: apiserver/facades/agent/secretsmanager/secrets.go (hunk 1 above)
Comment (jack-w-shaw) [reply apiserver/facades/agent/secretsmanager/secrets.go]: Yep you're right

**Pull request merged**